state/factory/statedb.go

```diff
@@ -267,11 +275,13 @@ func (sdb *stateDB) WorkingSet(ctx context.Context) (protocol.StateManager, erro
 	sdb.mutex.RLock()
 	height := sdb.currentChainHeight
 	sdb.mutex.RUnlock()
-	return sdb.newWorkingSet(ctx, height+1)
+	// TODO: the workingset should be read only, and should not be able to commit
```

Comment:
workingset should be able to writable which is requied by SimulateExecution, but unable to commit.